compat/src/main/scala-2.11_2.12/scala/collection/compat/PackageShared.scala

```diff
@@ -243,6 +250,40 @@ class TraversableExtensionMethods[A](private val self: c.Traversable[A]) extends
   def iterableFactory: GenericCompanion[Traversable] = self.companion
 }
 
+class TraversableLikeExtensionMethods[A, Repr](private val self: c.GenTraversableLike[A, Repr])
+    extends AnyVal {
+
+  def groupMap[K, B, VThat, That](key: A => K)(f: A => B)(
+      implicit traversable: IsTraversableLikeAux[A, Repr],
+      innerBf: CanBuildFrom[Repr, B, VThat],
+      outerBf: CanBuildFrom[c.GenMap[K, Repr], (K, VThat), That]): That = {
```

Comment:
I think we should fix the return type to `immutable.Map[K, VThat]`, like we do in the scala-library.

Reply:
I don't have a problem with that; however, it's not what `groupBy` does

Reply:
I was referring to [these implementations](https://github.com/scala/scala/blob/v2.13.1/src/library/scala/collection/Iterable.scala#L538-L630), I think that using as few `CanBuildFrom` as possible will make the code more readable :)

Reply:
I rewrote it to just use the 2.13 implementation, which massively simplified things